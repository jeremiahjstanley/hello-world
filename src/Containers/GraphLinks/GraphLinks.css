.graph-links {
	display: flex;
	justify-content: space-evenly;
}

.graph-links a {
	font-size: .75rem;
}