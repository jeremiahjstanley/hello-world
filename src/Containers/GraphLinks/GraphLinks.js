import React from 'react';
import { connect } from 'react-redux';
import { Link, withRouter } from 'react-router-dom'; 
import './GraphLinks.css'


export const GraphLinks = ({location, dataBase, dataSet, locationData}) => {

  return (
    <div className='graph-links'>
      <Link to='/stats/compare'>
        See how {location.name} stacks up against...
      </Link>
      <Link to='/stats/change_data_base'>
        Okay, that's {dataBase.name}, but what about?
      </Link>
      <Link to='/stats/change_data_set'>
        {dataSet.name} was cool right, explore other metrics?
      </Link>
      <Link to='/'>
        Start Over?
      </Link>
    </div>
  )
}

export const mapStateToProps = (state) => ({
  location: state.location,
  dataBase: state.dataBase,
  dataSet: state.dataSet,
});

export default withRouter(connect(mapStateToProps)(GraphLinks));