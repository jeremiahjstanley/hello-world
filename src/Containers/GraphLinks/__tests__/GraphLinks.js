import React from 'react';
import { shallow } from 'enzyme'
import { GraphLinks, mapStateToProps } from '../GraphLinks';


describe('GraphLinks', () => {

  let wrapper;
  let mockLocation;
  let mockDataBase;
  let mockDataSet;

  beforeEach(() => {

    mockLocation = [{
      name:'Burkina Faso',
      alpha_2:'BF',
      alpha_3:'BFA',
      country_code:'854',
      iso_3166_2:'ISO 3166-2: BF',
      region:'Africa',
      sub_region:'Sub-Saharan Africa',
      intermediate_region:'Western Africa',
      region_code:'002',sub_region_code:'202',
      intermediate_region_code:'011'
    }];

    mockDataSet = {
      name: 'Voice and Accountability',   
      dataset_code: 'VA'
    };

    mockDataBase = {
      name: 'Worldwide Governance Indicators',
      database_code: 'WWGI'
    };

    wrapper = shallow(<
    	GraphLinks
    		dataBase={mockDataBase}
    		dataSet={mockDataSet}
    		location={mockLocation}
    	/>
  	);

  });

	it('should match the snapshot', () => {

		expect(wrapper).toMatchSnapshot();

	});

	describe('mapStateToProps', () => {

		it('should return a state object with the keys of dataBase, dataSet, and location', () => {

      const mockState = {
        dataBase: mockDataBase,
        dataSet: mockDataSet,
        location: mockLocation
      };

      const expected = mockState;

      const mappedProps = mapStateToProps(mockState);

      expect(mappedProps).toEqual(expected);


		});
		
	});

});