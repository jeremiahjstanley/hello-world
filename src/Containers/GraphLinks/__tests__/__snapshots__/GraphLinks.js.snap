// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GraphLinks should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    dataBase={
        Object {
            "database_code": "WWGI",
            "name": "Worldwide Governance Indicators",
          }
    }
    dataSet={
        Object {
            "dataset_code": "VA",
            "name": "Voice and Accountability",
          }
    }
    location={
        Array [
            Object {
              "alpha_2": "BF",
              "alpha_3": "BFA",
              "country_code": "854",
              "intermediate_region": "Western Africa",
              "intermediate_region_code": "011",
              "iso_3166_2": "ISO 3166-2: BF",
              "name": "Burkina Faso",
              "region": "Africa",
              "region_code": "002",
              "sub_region": "Sub-Saharan Africa",
              "sub_region_code": "202",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="graph-link"
>
          <Link
                    replace={false}
                    to="/stats/compare"
          >
                    See how 
                    Burkina Faso
                     stacks up against...
          </Link>
</div>,
        <div
          className="graph-link"
>
          <Link
                    replace={false}
                    to="/stats/change_data_base"
          >
                    Okay, that's 
                    Worldwide Governance Indicators
                    , but what about?
          </Link>
</div>,
        <div
          className="graph-link"
>
          <Link
                    replace={false}
                    to="/stats/change_data_set"
          >
                    Voice and Accountability
                     was cool right, explore other metrics?
          </Link>
</div>,
      ],
      "className": "graph-links",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Link
            replace={false}
            to="/stats/compare"
>
            See how 
            Burkina Faso
             stacks up against...
</Link>,
          "className": "graph-link",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              "See how ",
              "Burkina Faso",
              " stacks up against...",
            ],
            "replace": false,
            "to": "/stats/compare",
          },
          "ref": null,
          "rendered": Array [
            "See how ",
            "Burkina Faso",
            " stacks up against...",
          ],
          "type": [Function],
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Link
            replace={false}
            to="/stats/change_data_base"
>
            Okay, that's 
            Worldwide Governance Indicators
            , but what about?
</Link>,
          "className": "graph-link",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              "Okay, that's ",
              "Worldwide Governance Indicators",
              ", but what about?",
            ],
            "replace": false,
            "to": "/stats/change_data_base",
          },
          "ref": null,
          "rendered": Array [
            "Okay, that's ",
            "Worldwide Governance Indicators",
            ", but what about?",
          ],
          "type": [Function],
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Link
            replace={false}
            to="/stats/change_data_set"
>
            Voice and Accountability
             was cool right, explore other metrics?
</Link>,
          "className": "graph-link",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              "Voice and Accountability",
              " was cool right, explore other metrics?",
            ],
            "replace": false,
            "to": "/stats/change_data_set",
          },
          "ref": null,
          "rendered": Array [
            "Voice and Accountability",
            " was cool right, explore other metrics?",
          ],
          "type": [Function],
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="graph-link"
>
            <Link
                        replace={false}
                        to="/stats/compare"
            >
                        See how 
                        Burkina Faso
                         stacks up against...
            </Link>
</div>,
          <div
            className="graph-link"
>
            <Link
                        replace={false}
                        to="/stats/change_data_base"
            >
                        Okay, that's 
                        Worldwide Governance Indicators
                        , but what about?
            </Link>
</div>,
          <div
            className="graph-link"
>
            <Link
                        replace={false}
                        to="/stats/change_data_set"
            >
                        Voice and Accountability
                         was cool right, explore other metrics?
            </Link>
</div>,
        ],
        "className": "graph-links",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Link
              replace={false}
              to="/stats/compare"
>
              See how 
              Burkina Faso
               stacks up against...
</Link>,
            "className": "graph-link",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                "See how ",
                "Burkina Faso",
                " stacks up against...",
              ],
              "replace": false,
              "to": "/stats/compare",
            },
            "ref": null,
            "rendered": Array [
              "See how ",
              "Burkina Faso",
              " stacks up against...",
            ],
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Link
              replace={false}
              to="/stats/change_data_base"
>
              Okay, that's 
              Worldwide Governance Indicators
              , but what about?
</Link>,
            "className": "graph-link",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                "Okay, that's ",
                "Worldwide Governance Indicators",
                ", but what about?",
              ],
              "replace": false,
              "to": "/stats/change_data_base",
            },
            "ref": null,
            "rendered": Array [
              "Okay, that's ",
              "Worldwide Governance Indicators",
              ", but what about?",
            ],
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Link
              replace={false}
              to="/stats/change_data_set"
>
              Voice and Accountability
               was cool right, explore other metrics?
</Link>,
            "className": "graph-link",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                "Voice and Accountability",
                " was cool right, explore other metrics?",
              ],
              "replace": false,
              "to": "/stats/change_data_set",
            },
            "ref": null,
            "rendered": Array [
              "Voice and Accountability",
              " was cool right, explore other metrics?",
            ],
            "type": [Function],
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
